common/hashing_test.cpp

```diff
@@ -288,53 +350,72 @@ auto FindBitRangeCollisions(llvm::ArrayRef<HashedValue<T>> hashes)
     bits_and_indices.push_back(
         {static_cast<uint32_t>(hash_bits), bits_and_indices.size()});
   }
+
+  // Now we sort by the extracted bit sequence so we can efficiently scan for
+  // colliding bit patterns.
   std::sort(
       bits_and_indices.begin(), bits_and_indices.end(),
       [](const auto& lhs, const auto& rhs) { return lhs.first < rhs.first; });
 
+  // Scan the sorted bit sequences we've extracted looking for collisions. We
+  // count the total collisions, but we also track the number of individual
+  // inputs that collide with each specific bit pattern.
   uint32_t prev_hash_bits = bits_and_indices[0].first;
   int prev_index = bits_and_indices[0].second;
   bool in_collision = false;
   int total = 0;
-  int distinct_collisions = 0;
   for (const auto& [hash_bits, hash_index] :
        llvm::ArrayRef(bits_and_indices).slice(1)) {
+    // Check if we've found a new hash (and thus a new value), reset everything.
     if (hash_bits != prev_hash_bits) {
+      CARBON_CHECK(hashes[prev_index].hash != hashes[hash_index].hash);
+      CARBON_CHECK(hashes[prev_index].v != hashes[hash_index].v);
       prev_hash_bits = hash_bits;
       prev_index = hash_index;
       in_collision = false;
       continue;
     }
 
+    // Otherwise, we have a colliding bit sequence.
     ++total;
+
+    // If we've already created a collision count to track this, just increment
+    // it and map this hash to it.
     if (in_collision) {
       ++collision_counts.back();
       collision_map[hash_index] = collision_counts.size() - 1;
       continue;
     }
+
+    // If this is a new collision, create a dedicated count to track it and
+    // begin counting.
     in_collision = true;
     collision_map[hash_index] = collision_counts.size();
     collision_counts.push_back(1);
-
-    ++distinct_collisions;
-    if (0 && distinct_collisions < 10) {
-      llvm::errs() << "Hash mask " << llvm::formatv("{0:x16}", BitMask)
-                   << " collision: " << hashes[prev_index] << " vs. "
-                   << hashes[hash_index] << "\n";
-    }
   }
 
-  // Sort by collisions.
+  // Sort by collision count for each hash.
   std::sort(bits_and_indices.begin(), bits_and_indices.end(),
             [&](const auto& lhs, const auto& rhs) {
               return collision_counts[collision_map[lhs.second]] <
                      collision_counts[collision_map[rhs.second]];
             });
 
+  // And compute the median and max.
   int median = collision_counts
       [collision_map[bits_and_indices[bits_and_indices.size() / 2].second]];
-  int max = collision_counts.back();
-  return {.total = total, .median_per_hash = median, .max_per_hash = max};
+  int max = *std::max_element(collision_counts.begin(), collision_counts.end());
+  return {.total = total, .median = median, .max = max};
+}
+
+auto CheckNoHashedDuplicates(llvm::ArrayRef<HashedString> hashes) -> void {
```

Comment:
```suggestion
auto CheckNoDuplicateValues(llvm::ArrayRef<HashedString> hashes) -> void {
```
or maybe `CheckNoDuplicateStrings`?